DiscordBot.DataAccess/EventsSheetsService.cs

```diff
@@ -297,25 +297,83 @@ public async Task ClearResponsesForUserAsync(int eventKey, ulong userId)
             await request.ExecuteAsync();
         }
 
-#pragma warning disable 1998 // Disable compiler warning stating that the unimplemented functions are synchronous
         public async Task<Dictionary<ulong, IEnumerable<EventResponse>>> GetSignupsByUserAsync(int eventId)
         {
-            return new Dictionary<ulong, IEnumerable<EventResponse>>
+            var request = sheetsService.Spreadsheets.Values.Get(
+                spreadsheetId,
+                $"{eventId}"
+            );
+            request.ValueRenderOption = GetRequest.ValueRenderOptionEnum.FORMATTEDVALUE;
+            var response = await request.ExecuteAsync();
+
+            if (response == null || response.Values.Count < 2)
             {
-                { 0UL, new[] { new EventResponse(":white_check_mark:", "Yes") } },
-                { 1UL, new[] { new EventResponse(":grey_question:", "Maybe") } }
-            };
+                throw new EventInitialisationException("Sign up sheet is empty");
+            }
+
+            var responseColumns = ParseResponseHeaders(response.Values);
+
+            return response.Values
+                .Skip(2)
+                .Select(row =>
+                {
```

Comment:
I'd recommend extracting this to its own method 🙂

Reply:
Removed in 6877d46 🙂